Chronos defaults include localhost as a staying node erroneously

#### Symptoms

Chronos thinks that localhost is staying in the cluster, if there are no other nodes which are staying in the cluster.

```
22-11-2016 15:29:44.283 UTC Status globals.cpp:226: Staying nodes:
22-11-2016 15:29:44.283 UTC Status globals.cpp:231:  - localhost:7253
22-11-2016 15:29:44.283 UTC Status globals.cpp:234: Joining nodes:
22-11-2016 15:29:44.283 UTC Status globals.cpp:239:  - 172.18.0.8
22-11-2016 15:29:44.283 UTC Status globals.cpp:255: Cluster view ID: 18015586077393156
22-11-2016 15:29:46.587 UTC Status alarm.cpp:62: chronos issued 3002.5 alarm
22-11-2016 15:29:47.534 UTC Error dnscachedresolver.cpp:846: Failed to retrieve record for localhost: Timeout while contacting DNS servers
22-11-2016 15:29:47.534 UTC Error httpclient.cpp:680: cURL failure with cURL error code 6 (see man 3 libcurl-errors) and HTTP error code 404
22-11-2016 15:29:47.534 UTC Warning chronos_internal_connection.cpp:470: Error response (404) to GET request to localhost:7253
22-11-2016 15:29:47.534 UTC Warning chronos_internal_connection.cpp:159: Resynchronisation with node localhost:7253 failed with rc 404
```

#### Impact

Chronos raises error logs at startup when creating a new cluster and may incorrectly resynchronize in some scenarios.

#### Release and environment 

Sprint Quilava

#### Steps to reproduce

Spin up a new Chronos with a single node in `chronos_cluster.conf` marked as joining and no nodes marked as staying. Check the chronos logs.

--- a/src/globals.cpp
+++ b/src/globals.cpp
@@ -61,9 +61,9 @@ Globals::Globals(std::string config_file,
   _desc.add_options()
     ("http.bind-address", po::value<std::string>()->default_value("0.0.0.0"), "Address to bind the HTTP server to")
     ("http.bind-port", po::value<int>()->default_value(7253), "Port to bind the HTTP server to")
-    ("cluster.localhost", po::value<std::string>()->default_value("localhost:7253"), "The address of the local host")
+    ("cluster.localhost", po::value<std::string>()->default_value("127.0.0.1:7253"), "The address of the local host")
     ("cluster.joining", po::value<std::vector<std::string>>()->multitoken()->default_value(std::vector<std::string>(), "HOST"), "The addresses of nodes in the cluster that are joining")
-    ("cluster.node", po::value<std::vector<std::string>>()->multitoken()->default_value(std::vector<std::string>(1, "localhost:7253"), "HOST"), "The addresses of nodes in the cluster that are staying")
+    ("cluster.node", po::value<std::vector<std::string>>()->multitoken()->default_value(std::vector<std::string>(), "HOST"), "The addresses of nodes in the cluster that are staying")
     ("cluster.leaving", po::value<std::vector<std::string>>()->multitoken()->default_value(std::vector<std::string>(), "HOST"), "The addresses of nodes in the cluster that are leaving")
     ("identity.instance_id", po::value<uint32_t>()->default_value(0), "A number between 0 and 127. The combination of instance ID and deployment ID should uniquely identify this node in the cluster, to remove the risk of timer collisions.")
     ("identity.deployment_id", po::value<uint32_t>()->default_value(0), "A number between 0 and 7. The combination of instance ID and deployment ID should uniquely identify this node in the cluster, to remove the risk of timer collisions.")
@@ -221,11 +221,21 @@ void Globals::update_config()
   std::vector<std::string> cluster_joining_addresses = conf_map["cluster.joining"].as<std::vector<std::string>>();
   set_cluster_joining_addresses(cluster_joining_addresses);
 
-  std::vector<std::string> cluster_staying_addresses = conf_map["cluster.node"].as<std::vector<std::string>>();
-  set_cluster_staying_addresses(cluster_staying_addresses);
-
   std::vector<std::string> cluster_leaving_addresses = conf_map["cluster.leaving"].as<std::vector<std::string>>();
   set_cluster_leaving_addresses(cluster_leaving_addresses);
+
+  std::vector<std::string> cluster_staying_addresses = conf_map["cluster.node"].as<std::vector<std::string>>();
+
+  // If there are no joining, staying or leaving addresses, add the local node
+  // to the staying nodes
+  if (cluster_staying_addresses.empty() &&
+      cluster_leaving_addresses.empty() &&
+      cluster_joining_addresses.empty())
+  {
+    cluster_staying_addresses.push_back(cluster_local_address);
+  }
+
+  set_cluster_staying_addresses(cluster_staying_addresses);
 
   // Figure out the new cluster by combining the nodes that are staying and the
   // nodes that are joining.

--- a/src/ut/test_globals.cpp
+++ b/src/ut/test_globals.cpp
@@ -80,12 +80,12 @@ TEST_F(TestGlobals, ParseGlobalsDefaults)
 
   std::string cluster_local_address;
   test_global->get_cluster_local_ip(cluster_local_address);
-  EXPECT_EQ(cluster_local_address, "localhost:7253");
+  EXPECT_EQ(cluster_local_address, "127.0.0.1:7253");
 
   std::vector<std::string> cluster_addresses;
   test_global->get_cluster_staying_addresses(cluster_addresses);
   EXPECT_EQ(cluster_addresses.size(), (unsigned)1);
-  EXPECT_EQ(cluster_addresses[0], "localhost:7253");
+  EXPECT_EQ(cluster_addresses[0], "127.0.0.1:7253");
 
   std::vector<std::string> cluster_joining_addresses;
   test_global->get_cluster_joining_addresses(cluster_joining_addresses);

